starship_module_config_derive/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+extern crate proc_macro;
+extern crate proc_macro2;
+
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, DeriveInput};
+
+#[proc_macro_derive(ModuleConfig)]
+pub fn derive_module_config(input: TokenStream) -> TokenStream {
+    let dinput = parse_macro_input!(input as DeriveInput);
+    impl_module_config(dinput)
+}
+
+fn impl_module_config(dinput: DeriveInput) -> TokenStream {
+    let struct_ident = &dinput.ident;
+    let (_impl_generics, ty_generics, where_clause) = dinput.generics.split_for_impl();
+
+    if let syn::Data::Struct(data) = dinput.data {
+        if let syn::Fields::Named(fields_named) = data.fields {
+            let mut tokens = quote! {};
+
+            for field in fields_named.named.iter() {
+                let ident = field.ident.as_ref().unwrap();
```

Comment:
Possible panic point. Can this happen when named fields does not have an identifier? Afaik, all fields in a struct has a name e.g. `symbol: &str`.